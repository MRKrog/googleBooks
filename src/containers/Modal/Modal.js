import React, { Component } from 'react';
import Book from '../Book/Book';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class Modal extends Component {
  render() {
    const { savedBooks, hideModal } = this.props;
    return (
      <div className="Modal">
        <h2>Your Saved Books</h2>
        <button className="hideModalBtn" onClick={hideModal}>X</button>
        <div className="BookDisplay">
          {
            savedBooks.map(singleBook => (
              <Book key={singleBook.id}
                    id={singleBook.id}
                    title={singleBook.title}
                    author={singleBook.author}
                    image={singleBook.image}
                    publisher={singleBook.publisher}
                    savedStatus={singleBook.savedStatus}
                />
              )
            )}
          </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  savedBooks: state.savedBooks,
})

Modal.propTypes = {
  savedBooks: PropTypes.array,
  hideModal: PropTypes.func,
}

export default connect(mapStateToProps, null)(Modal);
