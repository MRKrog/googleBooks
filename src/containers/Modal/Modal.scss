.Modal {
  display: flex;
  width: 100%;
  height: 100%;
  z-index: 9999;
  top: 0px;
  left: 0px;
  position: fixed;
  background: rgba(8, 8, 8, 0.97);
  padding: 2%;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  flex-direction: column;

  h2 {
    font-family: $fontOswald;
    font-size: 2em;
    color: #fff;
    font-weight: 100;
    margin: 0;
    text-transform: uppercase;
    padding-bottom: 10px;
  }

  .hideModalBtn {
    position: absolute;
    top: 0;
    right: 0;
    background: none;
    border: none;
    color: #fff;
    font-size: 2em;
    font-weight: 100;
    cursor: pointer;
    opacity: 1;
    transition: all .3s ease;

    &:hover {
      opacity: .8;
    }
  }
}
