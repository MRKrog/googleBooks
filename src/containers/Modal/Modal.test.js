import React, { Component } from 'react';
import { Modal, mapStateToProps } from './Modal';
import { shallow } from 'enzyme';

const mockSavedBooks = []

describe('Modal', () => {

  describe('Modal Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Modal />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

  })

  describe('mapStateToProps', () => {

  });

})
