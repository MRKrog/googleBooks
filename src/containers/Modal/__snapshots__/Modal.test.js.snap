// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Modal Component should match the snapshot 1`] = `
<div
  className="Modal"
>
  <h2>
    Your Saved Books
  </h2>
  <button
    className="hideModalBtn"
    onClick={[MockFunction]}
  >
    X
  </button>
  <div
    className="BookDisplay"
  >
    <Connect(Book)
      author="Cathy K. Kaufman"
      id="IFiIg7elnwIC"
      image="http://books.google.com/books/content?id=IFiIg7elnwIC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
      key="IFiIg7elnwIC"
      publisher="Greenwood Publishing Group"
      savedStatus={true}
      title="Cooking in Ancient Civilizations"
    />
  </div>
</div>
`;
